File: app/src/main/java/bj/discogsbrowser/model/artist/ArtistWantedUrl.java
package bj.discogsbrowser.model.artist;

import bj.discogsbrowser.artist.ArtistController;
import bj.discogsbrowser.epoxy.artist.UrlModel;

/**
 * Created by Josh Laird on 21/04/2017.
 * <p>
 * To ease {@link ArtistController}'s creation of {@link UrlModel}.
 */
public class ArtistWantedUrl
{
    private String url;
    private String displayText;
    private String hexCode;
    private String fontAwesomeString;

    public ArtistWantedUrl(String url, String displayText, String hexCode, String fontAwesomeString)
    {
        this.url = url;
        this.displayText = displayText;
        this.hexCode = hexCode;
        this.fontAwesomeString = fontAwesomeString;
    }

    public String getUrl()
    {
        return url;
    }

    public String getDisplayText()
    {
        return displayText;
    }

    public String getHexCode()
    {
        return hexCode;
    }

    public String getFontAwesomeString()
    {
        return fontAwesomeString;
    }
}

File: app/src/main/java/bj/vinylbrowser/model/artist/ArtistWantedUrl.kt
package bj.vinylbrowser.model.artist

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by Josh Laird on 19/05/2017.
 */
data class ArtistWantedUrl(val url: String,
                           val displayText: String,
                           val hexCode: String,
                           val fontAwesomeString: String) : Parcelable {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(url)
        dest.writeString(displayText)
        dest.writeString(hexCode)
        dest.writeString(fontAwesomeString)
    }

    override fun describeContents(): Int {
        return 0
    }
}